src/stackwalk.c

```diff
@@ -1165,7 +1181,15 @@ static void jl_rec_backtrace(jl_task_t *t) JL_NOTSAFEPOINT
 
 JL_DLLEXPORT void jl_gdblookup(void* ip)
 {
-    jl_print_native_codeloc((uintptr_t)ip);
+    char pre_str[64];
+    int ctid = -1;
+    pre_str[0] = '\0';
+    // do not call jl_threadid if there's no current task
+    if (jl_get_current_task()) {
+        ctid = jl_threadid() + 1;
+    }
+    snprintf(pre_str, 64, "thread (%d) ", ctid);
+    jl_print_native_codeloc(pre_str, (uintptr_t)ip);
```

Comment:
This may be more confusing than helpful. It is not guaranteed that the `ip` is currently executed on the current thread?